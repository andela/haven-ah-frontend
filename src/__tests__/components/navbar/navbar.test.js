import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { render, fireEvent, cleanup } from 'react-testing-library';
import Navbar from '../../../components/containers/navbar/navbar';

afterEach(cleanup);

const props = {
  isLoggedIn: false,
};

describe('Navbar', () => {
  it('should render without crashing', () => {
    render(<Router><Navbar {...props}/></Router>);
  });
});

describe('Navbar', () => {
  it('should contain header text', () => {
    const { getByText } = render(<Router><Navbar {...props}/></Router>);
    const header = getByText('Haven');
    const headerText = header.textContent;

    expect(headerText).toBe('Haven');
    expect(header).toMatchSnapshot();
  });

  it('should contain a clickable burger', () => {
    const handleClick = jest.fn();
    const { getByTestId } = render(<Router><Navbar {...props}/></Router>);
    const burger = getByTestId('burger');
    burger.onclick = handleClick;
    fireEvent.click(burger);

    expect(handleClick).toHaveBeenCalledTimes(1);
  });

  it('should match snapshot', () => {
    expect(Navbar).toMatchSnapshot();
  });
});
