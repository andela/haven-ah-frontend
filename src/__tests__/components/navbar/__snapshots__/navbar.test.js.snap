// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should contain header text 1`] = `
<span
  class="ah-orange"
>
  Haven
</span>
`;

exports[`Navbar should match snapshot 1`] = `[Function]`;
